redis/src/main/java/com/hazelcast/jet/contrib/redis/RedisSinks.java

```diff
@@ -0,0 +1,593 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.contrib.redis;
+
+import com.hazelcast.jet.function.FunctionEx;
+import com.hazelcast.jet.function.SupplierEx;
+import com.hazelcast.jet.pipeline.Sink;
+import com.hazelcast.jet.pipeline.SinkBuilder;
+import io.lettuce.core.LettuceFutures;
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.RedisFuture;
+import io.lettuce.core.RedisURI;
+import io.lettuce.core.ScoredValue;
+import io.lettuce.core.api.StatefulRedisConnection;
+import io.lettuce.core.api.async.RedisAsyncCommands;
+import io.lettuce.core.codec.RedisCodec;
+import io.lettuce.core.codec.StringCodec;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Collections.singletonMap;
+
+/**
+ * Contains factory methods for creating Redis sinks.
+ */
+public final class RedisSinks {
+
+    private RedisSinks() {
+    }
+
+    /**
+     * Creates a {@link Sink} which pushes data items into a specified Redis Hash. Assumes that all keys and values
+     * are {@link String}s and that incoming data items are raw {@link Map.Entry}s.
+     *
+     * @param name      name of the source being created
+     * @param uriString URI of the Redis server in String form
+     * @param hash      identifier of the Redis Hash being used
+     * @return sink to use in {@link com.hazelcast.jet.pipeline.Pipeline#drainTo}
+     */
+    public static Sink<Map.Entry<String, String>> hash(
```

Comment:
there are a lot of overloaded methods for each data structure
I think a full method and 1 or 2 conveniences for that method is enough most of the time
For example
we can get rid off either `hash(String name, String uri, String hash)` or `hash(RedisURI uri, String hash, String name)`